File: 프로그래머스/1/178871. 달리기 경주/달리기 경주.py
def solution(players, callings):
    position = {player : i for i, player in enumerate(players)}
    
    for c in callings:
        current_position = position[c]
        
        #딕셔너리 값 변경
        position[c] -= 1
        position[players[current_position - 1]] += 1
        
        #실제 값 변경
        players[current_position], players[current_position - 1] = players[current_position - 1], players[current_position]
        
    return players

File: 프로그래머스/1/178871. 달리기 경주/달리기 경주.js
function solution(players, callings) {
    var position = {};
    for (let i = 0; i < players.length; i++) {
        position[players[i]] = i;
    }
    
    for (let call of callings) {
        let current = position[call];
        let front_player = players[current - 1];
                            
        let temp = players[current];
        players[current] = players[current - 1];
        players[current - 1] = temp;
        
        position[call]--;
        position[front_player]++;
    }
    
    return players;
}